File: CS50x/sentimental-mario-more/mario.py
# import get_int from cs50
from cs50 import get_int


# define main
def main():
    height = get_height()
    start = 0
    for i in range(height):
        start += 1
        print(" " * (height-1), end="")
        print("#" * start, end="")
        print("  ", end="")
        print("#" * start)
        height -= 1


# define height function
def get_height():
    while True:
        try:
            # ask user for height input
            n = get_int("Height: ")
        except ValueError:
            print("Incorrect value")
        else:
            if n > 0 and n < 9:
                return n


main()

File: CS50x/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    //get input from user
    int size;
    do
    {
        size = get_int("Enter size of desired pyramids: ");
    }
    while (size < 1 || size > 8);

    //start building the pyramids
    int height = 0;
    while (height < size)
    {
        height++;
        //create front spaces
        int front_spaces = size;
        while (front_spaces > height)
        {
            printf(" ");
            front_spaces--;
        }

        //generate front blocks
        int front_blocks = 0;
        while (front_blocks < height)
        {
            printf("#");
            front_blocks++;
        }

        //center separator space
        printf("  ");

        //generate back blocks
        int back_blocks = 0;
        while (back_blocks < height)
        {
            printf("#");
            back_blocks++;
        }

        printf("\n");
    }
}
